src/org/pentaho/reporting/platform/plugin/output/StreamHtmlOutput.java

```diff
@@ -62,6 +63,9 @@ public int generate( final MasterReport report,
     }
     try {
       sp.processReport();
+      if ( listener != null && listener instanceof IAsyncReportListener ) {
```

Comment:
For this an all code like this: ReportListenerThreadHolder returns an IAsyncReportListener, so this check can be simplified by making the listener variable to be of type IAsyncReportListener .